backend/Functions/Edna.AssignmentLearnContent/Edna.AssignmentLearnContent/AssignmentLearnContentApi.cs

```diff
@@ -67,15 +81,36 @@ public AssignmentLearnContentApi(ILogger<AssignmentLearnContentApi> logger, IMap
         }
 
         [FunctionName(nameof(SaveAssignmentLearnContent))]
-        public void SaveAssignmentLearnContent(
+        public async Task<IActionResult> SaveAssignmentLearnContent(
             [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "assignments/{assignmentId}/learn-content/{contentUid}")] HttpRequest req,
-            [Table(AssignmentLearnContentTableName)] out AssignmentLearnContentEntity assignmentLearnContentEntity,
+            [Table(AssignmentLearnContentTableName)] IAsyncCollector<AssignmentLearnContentEntity> linksCollector,
             string assignmentId,
-            string contentUid)
+            string contentUid,
+            [Assignment(AssignmentId = "{assignmentId}")] Assignment assignment,
+            [Platform] PlatformsClient platformsClient,
+            [LtiAdvantage] INrpsClient nrpsClient,
+            [Lti1] Lti1MembershipClient membershipClient)
         {
+            IActionResult res = await ValidateUser(req, assignment, platformsClient, nrpsClient, membershipClient);
+            if (res.GetType() != typeof(OkResult))
+                return res;
+
             _logger.LogInformation($"Saving assignment learn content with uid [{contentUid}] to assignment {assignmentId}");
 
-            assignmentLearnContentEntity = new AssignmentLearnContentEntity { PartitionKey = assignmentId, RowKey = contentUid, ETag = "*" };
+            AssignmentLearnContentEntity assignmentLearnContentEntity = new AssignmentLearnContentEntity();
+            assignmentLearnContentEntity.PartitionKey = assignmentId;
+            assignmentLearnContentEntity.RowKey = contentUid;
+            assignmentLearnContentEntity.ETag = "*";
+
+            await linksCollector.AddAsync(assignmentLearnContentEntity);
+            await linksCollector.FlushAsync();
+
+            _logger.LogInformation("Learn Content saved.");

```

Comment:
We might consider removing these extraneous logs from the changes. Seeing them in Azure might not actually be of much help, or is there anything that I am missing w.r.t the importance of this log?

> Note: Let's do a quick pass of the code as well to see if we could remove logs which do not add any value. The best way to judge the log entry is to see what would we infer if it appears and if we could somehow infer the same thing in absence of it.